pkg/controller/dns/external_dns_test.go

```diff
@@ -98,7 +98,7 @@ func TestPublicConfig(t *testing.T) {
 				Subscription:       allZones.PublicZoneConfig.Subscription,
 				ResourceGroup:      allZones.PublicZoneConfig.ResourceGroup,
 				Provider:           manifests.PublicProvider,
-				DnsZoneResourceIDs: allZones.PublicZoneConfig.ZoneIds,
+				DnsZoneResourceIDs: util.Keys(allZones.PublicZoneConfig.ZoneIds),
```

Comment:
keys in maps in golang aren't ordered right? so this has the potential to be flaky?

wonder if sorting the keys function in utils is the best way to remedy inconsistency